### PR TITLE
Add Triggers Installation for e2e Tests

### LINKED ISSUE #1107: Install Triggers on Clusters Used for e2e Tests
### Feature request

In order to develop e2e tests for Triggers-related resources (e.g. EventListeners, TriggerTemplates, TriggerBindings, and ClusterTriggerBindings), the Triggers component must be installed on the clusters that tests run against. Currently, only pipelines is installed on our clusters.

### Use case

Developing e2e tests for the resources listed above.

### LINKED ISSUE #1115: e2e Test Fails to Get Correct PipelineRun Consistently
The pipeline start command for using `--pod-template` seems unable to correctly get the correct PipelineRun associated with the test. This issue seems related to the approach of getting the PipelineRun shown [here](https://github.com/tektoncd/cli/blob/ac555b84ece260d7a91c90d14279e75b7da30ff7/test/e2e/pipeline/pipeline_test.go#L298).

Due to the Pipeline having previous PipelineRuns associated with it, it seems a previous PipelineRun that succeeded is used instead of the PipelineRun created in the test that should fail. The PipelineRun that should fail in the test is failing as expected.

One approach may be to increase the [sleep time](https://github.com/tektoncd/cli/blob/ac555b84ece260d7a91c90d14279e75b7da30ff7/test/e2e/pipeline/pipeline_test.go#L296) used in the test to allow the PipelineRun to be created properly in time.

### PR DESCRIPTION
**Pull request by danielhelfand**
Closes #1107
Closes #1115 

Installing triggers on clusters used for e2e tests to allow for development of e2e tests using triggers resources.

Also fixes #1115 by adding option to sort TaskRuns/PipelineRuns in tests by start times to get correct runs associated with tests.

# Submitter Checklist

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: test/resources/eventlistener.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  name: listener
+spec:
+  serviceAccountName: tekton-triggers-example-sa
+  triggers:
+    - name: foo-trig
+      bindings:
+        - ref: pipeline-binding
+        - ref: message-binding
+      template:
+        name: pipeline-template
```
Comment (pratikjagrut) [inline test/resources/eventlistener.yaml:27]: I think we need a new line at the end.

Comment (danielhelfand) [reply test/resources/eventlistener.yaml:27]: Why? It's created successfully and verified by the tests added in this pr.

Comment (pratikjagrut) [reply test/resources/eventlistener.yaml:27]: I see the red circle with a dash in it, so I think it's because of the absence of \n.
Screenshot from 2020-08-08 13-31-42

**Comment by danielhelfand** [comment]
Should be ready for another look/should resolve some flakiness with e2e tests.

File: test/builder/builder.go [hunk 2]
```diff
@@ -171,6 +175,10 @@ func GetTaskRunListWithName(c *framework.Clients, tname string) *v1alpha1.TaskRu
 		log.Fatalf("Couldn't get expected taskRunList  %s", err)
 	}
 
+	if err == nil && sortByStartTime {
```
Comment (chmouel) [inline test/builder/builder.go:178]: did you mix some patch this addition doesn't seem to belong to this pr or have i missed something ?

Comment (danielhelfand) [reply test/builder/builder.go:178]: Yes, see #1115. Currently our e2e tests are not always using the correct PipelineRun or TaskRun due to them not being returned ordered by start time. Added to this pr as it was affecting it.
